src/drivers/deploy/refresh.sh.template

```diff
@@ -20,21 +20,46 @@
 pushd $(dirname "$0") > /dev/null
 
 echo "Re-label all the node in the cluster!"
-echo "New machine with GPU label will be labeled on kubernetes"
+echo "Keep pods consistent with labels."
 
-/bin/bash node-label.sh || exit $?
-
-echo "Remove the label on the machine without GPU tag"
+# "For NotIn rule, it's for Daemonset, so just update label is enough, Daemonset's status will change with labels."
 
 {% for host in machinelist %}
+    {% for role in serviceinfo['deploy-rules'] %}
+        
+        {% if 'in' in role %}    
+            {% if role['in'] in machinelist[host] %}
+                kubectl label --overwrite=true nodes {{ machinelist[ host ][ 'nodename' ] }} {{role['in']}}='true' || exit $?
+                if ! `kubectl get po -o wide | grep -1 {{machinelist[ host ]['nodename']}} | grep -q {{servicename}}`; then
+                    echo "Start service {{servicename}} frome Node {{ machinelist[ host ][ 'nodename'] }} according to the cluster-configuration label definithion change"
```

Comment:
Little confused, why if the service not on the node, we should re-execute the starting script?

Reply:
The label defined in config files, but service isn't up on the machine, maybe deleted by chance, so we should start the service on that labeled machine